### PR TITLE
go/cmd/vtbackup: report replication status metrics during catch-up phase

### LINKED ISSUE #13994: Feature Request: vtbackup, emit metric for catch-up replication errors
### Feature Description

Report a metric when `vtbackup` "catch-up replication" phase is in error state. For example, if the `pos` does not advance, or if replication stops:

 https://github.com/vitessio/vitess/blob/5342961f67974ef8b57c061574e53f41c4d30165/go/cmd/vtbackup/vtbackup.go#L444
https://github.com/vitessio/vitess/blob/5342961f67974ef8b57c061574e53f41c4d30165/go/cmd/vtbackup/vtbackup.go#L456

### Use Case(s)

If `vtbackup` replication breaks or stalls, currently there is no way to know except from logs. It would be nice if there were metrics we could alert on.

### PR DESCRIPTION
**Pull request by maxenglander**
## Description

Emit `vtbackup_phase_status` to report issues with individual phases of `vtbackup`. Specifically, during the catch-up phase of `vtbackup`, either of these metrics can be set to either `0` or `1`. At the beginning of `vtbackup` execution they are set to 0. If they are set to 1, that indicates a problem that may require manual intervention to resolve.

```
vtbackup_phase_status{phase="CatchUpReplication",status="Stalled"} 0
vtbackup_phase_status{phase="CatchUpReplication",status="Stopped"} 0
```

## Related Issue(s)

Fixes #13994

Have a metric test framework for `vtbackup` in https://github.com/vitessio/vitess/pull/12973, when that gets merged I can use that to test these new metrics.

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on the CI
-   [x] [Documentation](https://github.com/vitessio/website/pull/1585) was added or is not required

### CODE REVIEW

**Comment by maxenglander** [comment]
I've run unit tests locally. They are passing except for foreign key tests, which I think are probably not related to this PR.

**Comment by frouioui** [comment]
> I've run unit tests locally. They are passing except for foreign key tests (same as CI), which I think are probably not related to this PR.

@maxenglander The FK tests have been fixed in https://github.com/vitessio/vitess/pull/13997, you can rebase on `main` / merge `main` to see the fix.

File: go/cmd/vtbackup/vtbackup.go [hunk 1]
```diff
@@ -203,6 +218,13 @@ func main() {
 	topoServer := topo.Open()
 	defer topoServer.Close()
 
+	// Initialize stats.
+	for phaseName, statuses := range phaseStatuses {
+		for _, status := range statuses {
+			phaseStatus.Set([]string{phaseName, status}, int64(0))
```
Comment (ajm188) [inline go/cmd/vtbackup/vtbackup.go:224]: is the explicit `int64` cast necessary here?

File: go/cmd/vtbackup/vtbackup.go [hunk 2]
```diff
@@ -433,6 +455,9 @@ func takeBackup(ctx context.Context, topoServer *topo.Server, backupStorage back
 	backupParams.BackupTime = time.Now()
 
 	// Wait for replication to catch up.
+	var lastStatus replication.ReplicationStatus
+	var status replication.ReplicationStatus
+	var statusErr error
 	waitStartTime := time.Now()
```
Comment (ajm188) [inline go/cmd/vtbackup/vtbackup.go:461]: codestyle nit

```suggestion
	var (
		lastStatus replication.ReplicationStatus
		status replication.ReplicationStatus
		statusErr error

		waitStartTime = time.Now()
	)
```

**Review by ajm188 — APPROVED** [review]

File: go/cmd/vtbackup/vtbackup.go (hunk 1 above)
Comment (maxenglander) [reply go/cmd/vtbackup/vtbackup.go:224]: Nope! Fixed.

**Review by GuptaManan100 — APPROVED** [review]
LGTM!

**Pull request merged**